### PR TITLE
Close button for modals, plus onboarding text improvements

### PR DESCRIPTION
**Pull request by bionara**
# Description

Adds a close button to the modal, and implements it for covid test onboarding.
Also added a test for the onboarding close, and tweaked the onboarding button text to be next or OK on the last item.

Waiting for Swedish word for "Next" for "button-end" and "close" for "modal-close" (couldn't match it from existing se json)

## On what platform have you tested the change?

- [ ] Android - Emulator
- [ ] Android - Physical device
- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

image
image

## Checklist (for submission)

- [ ] Analytics/tracking events have been added (if relevant)
- [ ] Added new screen(s) to `src/features/screens.ts` and new Redux slice(s) to `mocks/mockedInitialState.ts` (if relevant)

## Checklist (for reviewers)

- [ ] Checked against Figma designs (if relevant)
- [ ] Checked that data has been successfully saved to the backend (if relevant)

## Out of scope and potential follow-up

[comment]: <> (Are there any related changes that you plan to do in a follow-up PR or voluntarily excluded from the scope?)

### CODE REVIEW

**Review by juliatan — CHANGES_REQUESTED** [review]
UI looks good Ben! I think you're missing the `await store.dispatch(fetchStartUpInfo());` dispatch action upon user hitting "close" though.

File: src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx [hunk 1]
```diff
@@ -79,6 +81,7 @@ export default function CovidTestListOnboardingModal(props: IProps) {
       swipeRight={previousModal}
       swipeLeft={nextModalOrComplete}
       testID={`covid-test-modal-screen-${onboardingModalScreenIndex}`}
+      showCloseButton={true}
```
[the remark this answers is not in the document]
Comment (bionara) [reply src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx:84]: done!

**Comment by bionara** [comment]
> UI looks good Ben! I think you're missing the `await store.dispatch(fetchStartUpInfo());` dispatch action upon user hitting "close" though.

It should actually be fine as the "close function" prop handles that:
https://github.com/zoe/covid-tracker-react-native/pull/1570/files#diff-f483ed13f83c85dfd7d0487b9efbe9288e5feba94e8a53ee91011582c28713fdR82

**Comment by juliatan** [comment]
> > UI looks good Ben! I think you're missing the `await store.dispatch(fetchStartUpInfo());` dispatch action upon user hitting "close" though.
> 
> It should actually be fine as the "close function" prop handles that:
> https://github.com/zoe/covid-tracker-react-native/pull/1570/files#diff-f483ed13f83c85dfd7d0487b9efbe9288e5feba94e8a53ee91011582c28713fdR82

Isn't that just a UI show/hide prop? Doesn't update backend I don't think..

**Comment by bionara** [comment]
> > > UI looks good Ben! I think you're missing the `await store.dispatch(fetchStartUpInfo());` dispatch action upon user hitting "close" though.
> > 
> > 
> > It should actually be fine as the "close function" prop handles that:
> > https://github.com/zoe/covid-tracker-react-native/pull/1570/files#diff-f483ed13f83c85dfd7d0487b9efbe9288e5feba94e8a53ee91011582c28713fdR82
> 
> Isn't that just a UI show/hide prop? Doesn't update backend I don't think..

yep by design that prop from onboarding calls the close on success. In a future PR we can make it close here if needed.

**Review by juliatan — APPROVED** [review]
Nice refactor !

File: src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx [hunk 2]
```diff
@@ -65,20 +58,34 @@ export default function CovidTestListOnboardingModal(props: IProps) {
     setIndexAndGif(selectedIndex);
   }
 
+  async function closeAndUpateStartupInfo() {
```
Comment (selvach) [inline src/features/covid-tests/modals/CovidTestListOnboardingModal.tsx:61]: NICE :)

**Pull request merged**